### PR TITLE
Improve plugin loading performance

### PR DESCRIPTION
**Pull request by Jack251970**
Improve plugin loading performance using concurrency

### CODE REVIEW

File: Flow.Launcher.Core/ExternalPlugins/Environments/AbstractPluginEnvironment.cs [hunk 1]
```diff
@@ -37,10 +38,10 @@ internal AbstractPluginEnvironment(List<PluginMetadata> pluginMetadataList, Plug
             PluginSettings = pluginSettings;
         }
 
-        internal IEnumerable<PluginPair> Setup()
```
Comment (taooceros) [inline Flow.Launcher.Core/ExternalPlugins/Environments/AbstractPluginEnvironment.cs:40]: why would you not use a generic version of task?

**Review by taooceros — COMMENTED** [review]
Could you give a small benchmark about the speed up?

File: Flow.Launcher.Core/ExternalPlugins/Environments/AbstractPluginEnvironment.cs (hunk 1 above)
Comment (Jack251970) [reply Flow.Launcher.Core/ExternalPlugins/Environments/AbstractPluginEnvironment.cs:40]: I have fixed this.

**Comment by Jack251970** [comment]
> Looks cool! Could you give a small benchmark about the speed up?

It seems this optimization is invalid. The instance-creating functions are executed very fast, and thus concurrency are hard to  optimize their performances.

**Pull request closed without merging**